tests/projects/test_projects.py

```diff
@@ -149,6 +149,7 @@ def test_run(tmpdir, tracking_uri_mock, use_start_run):  # pylint: disable=unuse
     for param in run.data.params:
         assert param.value == expected_params[param.key]
     expected_metrics = {"some_key": 3}
+    assert len(run.data.metrics) == len(expected_metrics)
```

Comment:
Verifies that metrics are logged to the correct FileStore from the blocking runs launched earlier. Previously we only checked that params were logged properly, which allowed the bug addressed by this PR to slip through the cracks as we log params prior to invoking the entry point command in the parent/driver process.